smarts/core/sensors.py

```diff
@@ -497,7 +497,9 @@ def _build_offscreen_camera(
         fb_props = FrameBufferProperties()
         fb_props.setRgbColor(True)
         fb_props.setRgbaBits(8, 8, 8, 1)
-        fb_props.setDepthBits(0)
+        # XXX: Though we don't need the depth buffer returned, setting this to 0
+        #      causes undefined behaviour where the ordering of meshes is random.
+        fb_props.setDepthBits(8)
```

Comment:
Did a 256 depth buffer fix the issue after all?

Reply:
Yes, I think so. I have attached the screenshots in the comment session.